qml/CoverPage.qml

```diff
@@ -22,9 +22,24 @@ import Sailfish.Silica 1.0
 CoverBackground {
     id: root
 
-    property bool update: status === Cover.Active || Qt.application.active
+    // This should be deprecated only use visibleChanged.
+    property bool updateCover: status === Cover.Active
+    onUpdateCoverChanged: {
+        if (updateCover) {
+            previewImage.updatePreview()
+        }
+    }
 
-    onUpdateChanged: previewImage.updatePreview()
+    // While peeking both main window and cover window are visible at the
+    // same time. Thus, we can capture from main window when cover
+    // becomes visible.
+    onVisibleChanged: {
+        if (visible) {
+            previewImage.updatePreview()
+        } else if (window.visible) {
+            previewImage.source = ""
```

Comment:
This also fixes grabbing of large pdf (my test pdf is >1.5MB) as the capture is taken when the cover is actually visible. You'll see on the cover what's rendered on the app window side. AFAIS the error is that we're trying to grab before the actual content of the document is rendered (documentItem changed signal is not enough).